physics/ephemeris.hpp

```diff
@@ -42,15 +42,52 @@ class Ephemeris {
   using NewtonianMotionEquation =
       SpecialSecondOrderDifferentialEquation<Position<Frame>>;
 
+  class AdaptiveStepParameters {
+   public:
+    // The |length_| and |speed_integration_tolerance|s are used to compute the
+    // |tolerance_to_error_ratio| for step size control.
+    AdaptiveStepParameters(
+        AdaptiveStepSizeIntegrator<NewtonianMotionEquation> const& integrator,
+        Length const& length_integration_tolerance,
+        Speed const& speed_integration_tolerance);
+
+    void SetTolerances(Length const& length_integration_tolerance,
+                       Speed const& speed_integration_tolerance);
+
+    template<typename T>
+    void WriteToMessage(not_null<T*> const t) const;
+
+    template<typename T>
+    static AdaptiveStepParameters ReadFromMessage(T const& t);
+
+   private:
+    // This will refer to a static object returned by a factory.
+    AdaptiveStepSizeIntegrator<NewtonianMotionEquation> const& integrator_;
```

Comment:
This makes the class non-assignable, which is a bit annoying for something that's just a parameter container.We tend to use pointers in those cases (the exception to our reference conventions is that we may take the address of a reference to a static object, this is also somewhat easier to enforce).
Make it a `not_null<AdaptiveStepSizeIntegrator<NewtonianMotionEquation>>`.